src/traces/heatmap/plot.js

```diff
@@ -137,10 +137,24 @@ function plotOne(gd, plotinfo, cd) {
     var imageWidth = Math.round(right - left),
         imageHeight = Math.round(bottom - top);
 
-    // now redraw
+    // setup image nodes
 
     // if image is entirely off-screen, don't even draw it
-    if(imageWidth <= 0 || imageHeight <= 0) return;
```

Comment:
`Heatmap.plot` was returning early for off screen traces, but did not clear the exiting nodes.